File: location_app/lib/models/property.dart
class Property {
  final String id;
  final String title;
  final String city;
  final int price;
  final int size;
  final List<String> features;
  final String ownerId;

  Property({
    required this.id,
    required this.title,
    required this.city,
    required this.price,
    required this.size,
    required this.features,
    required this.ownerId,
  });

  factory Property.fromJson(Map<String, dynamic> json) {
    return Property(
      id: json['_id'],
      title: json['title'],
      city: json['city'],
      price: json['price'],
      size: json['size'],
      features: List<String>.from(json['features'] ?? []),
      ownerId: json['ownerId'],
    );
  }
}

File: backend/models/Property.js
const mongoose = require('mongoose');

const propertySchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
  },
  images: [{ type: String }],
  city: String,
  price: Number,
  size: String,
  description: String,
  features: [String],
  ownerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
}, { timestamps: true });

module.exports = mongoose.model('Property', propertySchema);
